lib/scala/interpreter-dsl/src/main/java/org/enso/interpreter/dsl/impl/MethodProcessor.java

```diff
@@ -71,54 +71,64 @@ public boolean handleProcess(Set<? extends TypeElement> annotations, RoundEnviro
     return true;
   }
 
-  private void handleTypeElement(TypeElement element, Boolean needsFrame) throws IOException {
-    ExecutableElement executeMethod =
-        element.getEnclosedElements().stream()
-            .filter(
-                x -> {
-                  if (!(x instanceof ExecutableElement)) return false;
-                  Name name = x.getSimpleName();
-                  return name.contentEquals("execute");
-                })
-            .map(x -> (ExecutableElement) x)
-            .findFirst()
-            .orElseGet(
-                () -> {
-                  processingEnv
-                      .getMessager()
-                      .printMessage(Diagnostic.Kind.ERROR, "No execute method found.", element);
-                  return null;
-                });
-    if (executeMethod == null) return;
-    String pkgName =
-        processingEnv.getElementUtils().getPackageOf(element).getQualifiedName().toString();
-
-    MethodDefinition def = new MethodDefinition(pkgName, element, executeMethod, needsFrame);
-    if (!def.validate(processingEnv)) {
-      return;
+  private Element findExecuteMethod(TypeElement e) {
```

Comment:
Searches for `execute` method also in super classes, so it finds one provided by `Integer.BinaryNode`.